### ISSUE TITLE
thread  'biome::workspace_worker_x' has overflowed its stack

### ISSUE THREAD

**Issue by gc** [biomejs/biome#7728]
### Environment information

```block
CLI:
  Version:                      2.2.5
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.8.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.6.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

It appears that a corrupt/invalid JSON file (of a sufficient size) causes the 'has overflowed its stack' error. Obviously, I shouldn't have a bad json file laying around, but I accidentally created it and didn't know it was there, ideally it would be handled more gracefully. 

**Minimal Reproduction** (https://github.com/gc/biome-crash)

This is a barebones biome setup, with `evil.json` causing the crash, alternatively if you just copy this file to somewhere you run biome that should work. The playground doesn't like me trying to paste in this text (it hangs badly)

```sh
git clone https://github.com/gc/biome-crash.git

cd biome-crash

npm install

npm run lint (Runs biome check)
```
```
> biome check --log-level=info

  2025-10-11T00:59:19.239702Z  INFO  Configuration file loaded: Some("C:\\dev\\biome-crash\\biome.json"), diagnostics detected 0
    at crates\biome_cli\src\commands\mod.rs:900 on main

  2025-10-11T00:59:19.243489Z  INFO  Pulled 0 diagnostic(s), skipped 0 diagnostic(s) from C:\dev\biome-crash\biome.json
    at crates\biome_service\src\workspace\server.rs:1342 on biome::workspace_worker_3

  2025-10-11T00:59:19.243596Z  INFO  Pulled 0 diagnostic(s), skipped 0 diagnostic(s) from C:\dev\biome-crash\biome.json
    at crates\biome_service\src\workspace\server.rs:1342 on biome::workspace_worker_3

  2025-10-11T00:59:19.243672Z  INFO  Pulled 0 diagnostic(s), skipped 0 diagnostic(s) from C:\dev\biome-crash\src\good.json
    at crates\biome_service\src\workspace\server.rs:1342 on biome::workspace_worker_4

  2025-10-11T00:59:19.243674Z  INFO  Pulled 0 diagnostic(s), skipped 0 diagnostic(s) from C:\dev\biome-crash\package.json
    at crates\biome_service\src\workspace\server.rs:1342 on biome::workspace_worker_5

  2025-10-11T00:59:19.243850Z  INFO  Pulled 0 diagnostic(s), skipped 0 diagnostic(s) from C:\dev\biome-crash\package.json
    at crates\biome_service\src\workspace\server.rs:1342 on biome::workspace_worker_5

  2025-10-11T00:59:19.243762Z  INFO  Pulled 0 diagnostic(s), skipped 0 diagnostic(s) from C:\dev\biome-crash\src\good.json
    at crates\biome_service\src\workspace\server.rs:1342 on biome::workspace_worker_4


thread 'biome::workspace_worker_0' has overflowed its stack
```

Note that `evil.json` doesn't show up in the output at all, confusing me into thinking `good.json` was the culprit.

### Expected result

To not crash, and instead skip the file and continue on, OR to more gracefully handle the error then exit/stop.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by gc** [comment]
Possibly related to https://github.com/biomejs/biome/issues/6850

**Comment by John-Colvin** [comment]
I hit this as well. In my case it was caused by moving all my exclusions (`!` patterns in `files.includes`) above `**`. Once I reverted to the order `"**", <all the exclude patterns>` it stopped happening.

**Comment by mecaota** [comment]
I also encountered the same problem. It seems there are multiple patterns to this bug, and I suspect it might be due to changing the file names from camel case to Pascal case to standardize capitalization, but I couldn't create a reproducible repository.

What I can say for sure is that the problem was resolved by deleting all the files in the repository and then restoring them again using `git checkout`. The same applies if you manually add files that are not under git control, such as environment files.

**Comment by dyc3** [comment]
At a glance, the crash is happening when it's trying to drop a bunch of (assuming bogus) syntax nodes in `Rc<>`. My current theory is that the JSON parser in particular handles bogus nodes in some way that causes it to create a really deep tree of nodes, and trying to free all of them is causing the stack to blow up because it does it recursively.

**Comment by simon-the-shark** [comment]
yeah `git clean -fdx` did the job
